libs/map/src/components/legend/vl-map-legend.ts

```diff
@@ -152,29 +192,57 @@ export class VlMapLegend extends BaseLitElement {
         return new URL(`?${legendSearchParams}`, layerUrl);
     }
 
+    private customLegendItems(): CustomItem[] {
+        const childNodeList = Array.from(this.childNodes);
+        const legendItemsList = childNodeList.filter((child) => child instanceof VlMapLegendItem);
+        const legendItems: CustomItem[] = legendItemsList.map((item: VlMapLegendItem) => {
+            return { type: 'custom', name: item.layer, styleElement: item.cloneNode(true) as VlMapLegendItem };
+        });
+        return legendItems;
+    }
+
     private updateLegendGeometryItems(layers: VlMapLayer[]) {
+        let items: StyledItem[] = [];
         layers.forEach((layer) => {
             if (layer._styles.length === 1) {
                 const style = layer._styles[0];
 
```

Comment:
met betrekking tot opmerking die je gisteren gaf en die van Pascal;
op het eerste zicht lijkt het een refactor maar blijft het functioneel hetzelfde?

Reply:
De functionaliteit is hetzelfde alleen de bug dat bij meerdere styles per laag de styles van andere lagen overschreven werden is aangepast.

Reply:
excellent!